cprover_bindings/src/goto_program/expr.rs

```diff
@@ -1493,6 +1493,10 @@ impl Expr {
         Stmt::assign(self, rhs, loc)
     }
 
+    pub fn deinit(self, loc: Location) -> Stmt {
```

Comment:
We definitely want a comment here, especially as "deinit" is not a c notion really, and this is "cprover_bindings"

That's kind of why I thought this should be called "poison" but I don't remember if that was decided against for some reason?

Also, I don't know if this is me reading it weird, but I have the comment note that this is a shorthand/helper (perhaps by example? Like: "Construct a deinit statement from an expression by writing `expr.deinit()`"

Reply:
I think it's because @celinval (correct me if I'm wrong) pointed out that "poison" in this form is only intended to model uninitialized memory. So calling it poison is too vague and deinit makes more sense. I tend to agree with that